backend/src/main/java/de/robinschatzl/gameapp/backend/security/UserController.java

```diff
@@ -41,4 +44,17 @@ public void logout(HttpSession httpSession) {
         SecurityContextHolder.clearContext();
     }
 
+    @PostMapping("/signup")
+    //eslint-disable-next-line
+    public MongoUser signUp(@RequestBody @Valid MongoUser user) {
+        String errorMessage = "Username already exists!";
+
+        if (mongoUserRepository.findMongoUserByUsername(user.username()).isPresent()) {
+            throw new IllegalArgumentException(errorMessage);
+        }
+        String encodedPassword = passwordEncoder.encode(user.password());
+        MongoUser newUser = new MongoUser(null, user.username(), encodedPassword);
+        return mongoUserRepository.save(newUser);
+    }
```

Comment:
you could bring the errorMessage inside the if: it's better because you only initialize it, when you need id.
```suggestion
    public MongoUser signUp(@RequestBody @Valid MongoUser user) {
        if (mongoUserRepository.findMongoUserByUsername(user.username()).isPresent()) {
            String errorMessage = "Username already exists!";
            throw new IllegalArgumentException(errorMessage);
        }
        String encodedPassword = passwordEncoder.encode(user.password());
        MongoUser newUser = new MongoUser(null, user.username(), encodedPassword);
        return mongoUserRepository.save(newUser);
    }
```

Reply:
ahh THX works fine